ssh-server/main.go

```diff
@@ -113,38 +118,99 @@ func serviceChannel(newChannel ssh.NewChannel) error {
 	}
 	defer channel.Close()
 
+	allow := map[string]bool{"shell": true, "pty-req": true, "window-change": true}
+	resize := make(chan *pty.Winsize, 1)
+	defer close(resize)
+	done := make(chan struct{})
+	for {
+		select {
+		case <-done:
+			return nil
+		case req, ok := <-requests:
+			if !ok {
+				return nil
+			}
+			if req.WantReply {
+				req.Reply(allow[req.Type], nil)
+			}
+			switch req.Type {
+			case "pty-req":
+				r := bytes.NewReader(req.Payload)
+				term, err := readString(r)
+				if err != nil {
+					return fmt.Errorf("parsing TERM environment variable value: %v", err)
+				}
+
+				size, err := readWindowSize(r)
```

Comment:
I'm still not convinced the asymmetry in error handling between the initial resize and other resizes makes sense.  Do you actually expect failures?

Reply:
I've never seen failures on resize. I've changed it to handle them the same.